cloud/pkg/devicecontroller/devicecontroller.go

```diff
@@ -45,22 +58,13 @@ func (dc *DeviceController) Enable() bool {
 
 // Start controller
 func (dc *DeviceController) Start() {
-	downstream, err := controller.NewDownstreamController()
-	if err != nil {
-		klog.Fatalf("New downstream controller failed with error: %s", err)
-	}
-	upstream, err := controller.NewUpstreamController(downstream)
-	if err != nil {
-		klog.Fatalf("new upstream controller failed with error: %s", err)
-	}
-
-	if err := downstream.Start(); err != nil {
+	if err := dc.downstream.Start(); err != nil {
 		klog.Fatalf("start downstream failed with error: %s", err)
 	}
 	// wait for downstream controller to start and load deviceModels and devices
 	// TODO think about sync
```

Comment:
`TODO think about sync` means wait for devciceinformer.HasSynced and devciemodelinformer.HasSynced method?

Reply:
Looks like it's need to wait all deviceModels and devices cached(cache here maybe also can use the informer, instead of define a new one in devicecontroler) in devicecontroller, `HasSynced`  should be a efficient way, we can do series refactor in folowup prs.